packages/spectacle/src/utils/remark-rehype-directive.ts

```diff
@@ -0,0 +1,96 @@
+// @ts-ignore
+import unistVisit, { Visitor } from 'unist-util-visit';
+import { Literal, Node, Parent } from 'unist';
+import * as mdast from 'mdast-builder';
+import type { Plugin } from 'unified';
+import visit from 'unist-util-visit';
+import cloneDeep from 'lodash.clonedeep';
+
+const directiveNodeTypes = ['sectionDirective'] as const;
+export const directiveMatch = /^::(.*)$/;
+
+type DirectiveNodes = typeof directiveNodeTypes[number];
+type ValueNode = Node & { value: string; children?: unknown[] };
+type VisitNode = Node & {
+  name: string;
+  attributes?: Record<string, string> | undefined;
+  children?: unknown[];
+};
+type TreeNode = Node & { children: Node[] };
+
+const cloneFn =
+  typeof structuredClone !== 'undefined' ? structuredClone : cloneDeep;
```

Comment:
this doesn't always exist and it appears in our instance `cloneDeep` from lodash is an acceptable fill-in.